src/widgets/immunizations/immunizations-form.component.tsx

```diff
@@ -0,0 +1,348 @@
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+  const formRef = useRef<HTMLFormElement>(null);
+  const { t } = useTranslation();
+  const history = useHistory();
+  const today = new Date().toISOString().split("T")[0];
+  const currentUser = useSessionUser();
+
+  useEffect(() => {
+    if (vaccinationDate) {
+      setEnableCreateButtons(true);
+    } else {
+      setEnableCreateButtons(false);
+    }
+  }, [vaccinationDate]);
+
+  useEffect(() => {
+    if (viewEditForm && formChanged) {
+      setEnableEditButtons(true);
+    } else {
+      setEnableEditButtons(false);
+    }
+  }, [viewEditForm, formChanged]);
+
+  useEffect(() => {
+    if (props.match.params) {
+      const {
+        immunizationObsUuid,
+        vaccineName,
+        vaccineUuid,
+        manufacturer,
+        expirationDate,
+        vaccinationDate,
+        lotNumber,
+        sequences,
+        currentDose
+      }: Immunization = props.match.params[0];
+
+      setImmunizationObsUuid(immunizationObsUuid);
+      setVaccineName(vaccineName);
+      setVaccineUuid(vaccineUuid);
+      setManufacturer(manufacturer);
+      setVaccinationDate(vaccinationDate);
+      setVaccinationExpiration(expirationDate);
+      setLotNumber(lotNumber);
+      setViewEditForm(false);
+      if (hasSequences(sequences)) {
+        setImmunizationSequences(sequences);
+      }
+      if (vaccineName && vaccinationDate) {
+        setViewEditForm(true);
+        if (hasSequences(sequences)) {
+          setCurrentDose(currentDose);
+        }
+      }
+    }
+  }, [props.match.params]);
+
+  const handleFormSubmit = event => {
+    event.preventDefault();
+    const currentVisitUuid = getStartedVisit?.getValue()?.visitData?.uuid;
+    const currentLocationUuid = currentUser?.sessionLocation?.uuid;
+    const currentProviderUuid = currentUser?.currentProvider?.uuid;
+
+    const immunization: Immunization = {
+      patientUuid: patientUuid,
+      immunizationObsUuid: immunizationObsUuid,
+      vaccineName: vaccineName,
+      vaccineUuid: vaccineUuid,
+      manufacturer: manufacturer,
+      expirationDate: vaccinationExpiration,
+      vaccinationDate: vaccinationDate,
+      lotNumber: lotNumber,
+      currentDose: currentDose,
+      sequences: immunizationSequences
+    };
+    const abortController = new AbortController();
+
+    savePatientImmunization(
+      mapToFhirImmunizationResource(
+        immunization,
+        currentVisitUuid,
+        currentLocationUuid,
+        currentProviderUuid
+      ),
+      patientUuid,
+      immunizationObsUuid,
+      abortController
+    ).then(response => {
+      response.status == 200 && navigate();
+    }, createErrorHandler);
+    return () => abortController.abort();
+  };
+
+  function navigate() {
+    history.push(`/patient/${patientUuid}/chart/immunizations`);
+    props.closeComponent();
+  }
+
+  function createForm() {
+    const addFormHeader = t("add vaccine", "Add Vaccine") + ": " + vaccineName;
```

Comment:
The `vaccineName` should rather be an argument for the localization. The `{label}: {value}` is not applicable in all languages, in some you may want something different from a colon, in others you may want to switch order.